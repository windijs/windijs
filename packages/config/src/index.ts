export * from "./bg";
export * from "./svg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./effects";
export * from "./behaviors";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./animation";
export * from "./transform";
export * from "./positioning";
export * from "./interactivity";
export * from "./transitions";
export * from "./typography";
export * from "./keyframes";
export * from "./interface";
