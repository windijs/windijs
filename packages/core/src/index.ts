export * from "./base";
export * from "./api";
export * from "./handler";
export * from "./prefixer";
export * from "./builder";
