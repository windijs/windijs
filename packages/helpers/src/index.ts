// TODO: maybe add more helpers reference sass?
// https://sass-lang.com/documentation/modules

export * from "./types";
export * from "./build";
export * from "./config";
export * from "./color";
export * from "./common";
export * from "./css";
export * from "./data";
export * from "./dollar";
export * from "./escape";
export * from "./funcs";
export * from "./math";
export * from "./meta";
export * from "./namer";
export * from "./proxy";
export * from "./unit";
