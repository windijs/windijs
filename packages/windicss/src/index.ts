export * from "./vm";
export * from "./processor";
