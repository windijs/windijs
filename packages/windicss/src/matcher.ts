// TODO: maybe using unocss matcher?

export function match () {

}
