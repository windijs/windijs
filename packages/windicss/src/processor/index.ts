export * from "./common";
export * from "./utility";
export * from "./arbitrary";
