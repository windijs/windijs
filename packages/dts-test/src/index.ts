import { defineConfig } from "@windijs/config";

defineConfig({
  utilities: {

  },
});
