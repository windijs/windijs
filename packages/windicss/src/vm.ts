import { Expression, Parser, ParserOptions, Value } from "expr-eval";

import { StyleObject } from "@windi/helpers";

type WindiExpression = Expression & { eval(ctx?: Value | undefined): StyleObject[] };

export class Vm<T extends object = {}> extends Parser {
  ctx: T;

  constructor (options?: ParserOptions, ctx?: T) {
    super(options);
    this.ctx = ctx ?? {} as T;
  }

  setContext (ctx: T) {
    this.ctx = ctx;
  }

  parse (src: string): WindiExpression {
    const expr = super.parse("[" + src + "]");

    Reflect.defineProperty(expr, "eval", {
      value: (ctx?: Value | undefined) => {
        return expr.evaluate({ ...this.ctx as object, ...ctx as object });
      },
    });

    return expr as WindiExpression;
  }
}
