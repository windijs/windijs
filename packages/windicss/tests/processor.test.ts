import { UtilityExtractor, UtilityArbitraryExtractor, StyleArbitraryExtractor, Processor } from "../src";

import * as utilities from "@windijs/utilities";
import * as variants from "@windijs/variants";
import { style } from "@windijs/style";

const processor = new Processor({
  separator: "-",
  variantSeparator: ":",
  utilities: {
    ...utilities,
    style,
  },
  variants,
  extractors: [UtilityExtractor, UtilityArbitraryExtractor, StyleArbitraryExtractor],
});

test("interpret", () => {
  expect(
    processor.interpret(
      "bg-red-500 hover:focus:bg-blue-400 rounded style-backgroundColor-red style-padding-1rem [padding:1rem] [font-size:14em] p-[15px] bg-[#1c1c1e] bg-opacity-[0.25]"
    )
  ).toMatchSnapshot();
});

test("compile", () => {
  expect(
    processor.compile(
      "bg-red-500 hover:focus:bg-blue-400 rounded style-backgroundColor-red style-padding-1rem [padding:1rem] [font-size:14em] p-[15px] bg-[#1c1c1e] bg-opacity-[0.25]"
    )
  ).toMatchSnapshot();
});
