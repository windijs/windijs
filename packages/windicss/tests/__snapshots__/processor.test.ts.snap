// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile 1`] = `
".windi-yghcn7 {
  background-color: rgba(28, 28, 30, var(--w-bg-opacity));
  --w-bg-opacity: 0.25;
}

.windi-yghcn7:hover:focus {
  background-color: rgba(96, 165, 250, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.windi-yghcn7 {
  border-radius: 0.25rem;
  padding: 1rem;
  font-size: 14em;
}"
`;

exports[`interpret 1`] = `
".bg-red-500 {
  background-color: rgba(239, 68, 68, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.hover\\\\:focus\\\\:bg-blue-400:hover:focus {
  background-color: rgba(96, 165, 250, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.rounded {
  border-radius: 0.25rem;
}

.style-backgroundColor-red {
  background-color: red;
}

.style-padding-1rem {
  padding: 1rem;
}

.p-\\\\[15px\\\\] {
  padding: 15px;
}

.bg-\\\\[\\\\#1c1c1e\\\\] {
  background-color: rgba(28, 28, 30, var(--w-bg-opacity));
  --w-bg-opacity: 1;
}

.bg-opacity-\\\\[0\\\\.25\\\\] {
  --w-bg-opacity: 0.25;
}

.\\\\[padding\\\\:1rem\\\\] {
  padding: 1rem;
}

.\\\\[font-size\\\\:14em\\\\] {
  font-size: 14em;
}"
`;
